push/src/main/java/org/aerogear/mobile/push/UnifiedPushConfig.java

```diff
@@ -22,7 +29,29 @@ public void setAlias(String alias) {
     }
 
     public void setCategories(List<String> categories) {
+        validateCategories(categories);
         this.categories = new ArrayList<>(categories);
     }
 
+
+    /**
+     * Validates categories against Google's pattern.
+     *
+     * @param categories a group of Strings each will be validated.
+     * @throws IllegalArgumentException if a category fails to match [a-zA-Z0-9-_.~%]{1,900}
+     */
+    private static void validateCategories(List<String> categories) {
```

Comment:
If this method would return the `categories` object, we could change the `setCategories` method to a one liner.

Moreover, if `null` is a valid _list of categories_ if `categories` is null, it could return a `Collections.emptyList()`: that way we could avoid the `null` check (mandatory to invoke `new ArrayList<>(categories)`

Reply:
I don't like to have validation methods returning anything